.changeset/red-rings-tell.md

```diff
@@ -0,0 +1,38 @@
+---
+'astro': patch
+---
+
+Adds additional validation options to `astro:env`
+
+`astro:env` schema datatypes `string` and `number` now have new optional validation rules:
+
+```js
+import { defineConfig, envField } from 'astro/config'
+
+export default defineConfig({
+    experimental: {
+        env: {
+            schema: {
+                FOO: envField.string({
+                    // ...
+                    max: 32,
+                    min: 3,
+                    length: 12,
+                    url: true,
+                    includes: 'foo',
+                    startsWith: 'bar',
+                    endsWith: 'baz'
```

Comment:
I guess the reason we can't have a `check(value: string) => boolean` sort of thing is because this needs to happen at runtime, right?

Reply:
Yeah exactly, everything has to be serializable